### ISSUE TITLE
Individual column searching (text inputs and selects)

### ISSUE THREAD

**Issue by mauthi** [laravel-enso/DataTable#13]
It would be nice if you offer an option to filter individual columns like described here:
https://datatables.net/examples/api/multi_filter_select.html

Maybe you can give me an advice where to implement it and I'll try to add the funcunality?

**Comment by aocneanu** [comment]
TL;DR:

You can use the `extraFilters` and accomplish exactly that.

I'm not a big fan of individual input filtering. We find that one the most useful feature of this package is the single search input that can have multiple arguments and search everywhere... This is also one of our main "market differentiators" :)

Detailed story:

Both `extraFilters` and `intervalFilters` have a watcher and every time something is changed in them the table will reload. You don't even have to worry about debouncing because it's globally applied on the method that gets the data from server.

This means that right now you can do something like this:

define in `data`
```
"extraFilters": {
    "orders": {
        "number" : '',
        "field_2" : '',
    }
}
```

and in the template
```
<input type="text" v-model="extraFilters.orders.number" placeholder="Search by order number">

<data-table id="orders" source"/orders" :extra-filters="extraFilters"></data-table>
```

and now you can filter by order number within the results that already displayed.

If you do that I recommend that you remove the `order.number` column from the searchable array in the `TableStructureClass`: `"notSearchable": [columnIndex]"`, because it should be useless to filter by order number from two places, and you don't want to waste resources.

Besides the above, we have a custom filter that was not yet migrated from the old adminlte but will be in the next days. It can be used combined with the custom inputs, and with `vue-select` too, and the result can be something like:

![image](https://user-images.githubusercontent.com/16073274/32737565-1a8b6c66-c8a3-11e7-90ab-97e1fd727118.png)


Cheers.

**Comment by aocneanu** [comment]
Meanwhile in laravel-enso/VueComponents@9add3067f9886019d6191032940e198452841426

![image](https://user-images.githubusercontent.com/16073274/32741738-f220ac2a-c8af-11e7-9120-1a816baddbd2.png)

**Comment by mauthi** [comment]
Thx for the description abvor, works fine.
Also thx for merging the component.
In addition to that it would be nice if you can provide a short description how to use this component (here or better in documentation) - thx
